tests/integrated/test-drift-instability/2fluid.cxx

```diff
@@ -3,415 +3,420 @@
-
-  if(evolve_te) {
-    bout_solve(Te, "Te");
-    comms.add(Te);
-    output.write("te\n");
-  }else
-    initial_profile("Te", Te);
-
-  if(evolve_ajpar) {
-    bout_solve(Ajpar, "Ajpar");
-    comms.add(Ajpar);
-    output.write("ajpar\n");
-  }else {
-    initial_profile("Ajpar", Ajpar);
-    if (ZeroElMass) {
-      SAVE_REPEAT(Ajpar); // output calculated Ajpar
+    BoutReal hthe0;
+    if (mesh->get(hthe0, "hthe0") == 0) {
+      output.write(
+          "    ****NOTE: input from BOUT, Z length needs to be divided by {:e}\n",
+          hthe0 / rho_s);
     }
-  }
 
-  if(evolve_vi) {
-    bout_solve(Vi, "Vi");
-    comms.add(Vi);
-    output.write("vi\n");
-  }else
-    initial_profile("Vi", Vi);
-
-  if(evolve_ti) {
-    bout_solve(Ti, "Ti");
-    comms.add(Ti);
-    output.write("ti\n");
-  }else
-    initial_profile("Ti", Ti);
-  
-  // Set boundary conditions
-  jpar.setBoundary("jpar");
-
-  /************** SETUP COMMUNICATIONS **************/
-
-  // add extra variables to communication
-  comms.add(phi);
-  comms.add(Apar);
-
-  // Add any other variables to be dumped to file
-  SAVE_REPEAT(phi, Apar, jpar);
-  SAVE_ONCE(Ni0, Te0, Ti0, Te_x, Ti_x, Ni_x, rho_s, wci);
-
-  if (mesh->StaggerGrids) {
-    maybe_ylow = CELL_YLOW;
-  } else {
-    maybe_ylow = CELL_CENTRE;
+    /************** NORMALISE QUANTITIES *****************/
+
+    output.write("\tNormalising to rho_s = {:e}\n", rho_s);
+
+    // Normalise profiles
+    Ni0 /= Ni_x / 1.0e14;
+    Ti0 /= Te_x;
+    Te0 /= Te_x;
+    phi0 /= Te_x;
+    Vi0 /= Vi_x;
+
+    // Normalise curvature term
+    b0xcv.x /= (bmag / 1e4);
+    b0xcv.y *= rho_s * rho_s;
+    b0xcv.z *= rho_s * rho_s;
+
+    // Normalise geometry
+    Rxy /= rho_s;
+    hthe /= rho_s;
+    I *= rho_s * rho_s * (bmag / 1e4) * ShearFactor;
+    coord->dx /= rho_s * rho_s * (bmag / 1e4);
+
+    // Normalise magnetic field
+    Bpxy /= (bmag / 1.e4);
+    Btxy /= (bmag / 1.e4);
+    coord->Bxy /= (bmag / 1.e4);
+
+    // calculate pressures
+    pei0 = (Ti0 + Te0) * Ni0;
+    pe0 = Te0 * Ni0;
+
+    /**************** CALCULATE METRICS ******************/
+
+    coord->g11 = SQ(Rxy * Bpxy);
+    coord->g22 = 1.0 / SQ(hthe);
+    coord->g33 = SQ(I) * coord->g11 + SQ(coord->Bxy) / coord->g11;
+    coord->g12 = 0.0;
+    coord->g13 = -I * coord->g11;
+    coord->g23 = -Btxy / (hthe * Bpxy * Rxy);
+
+    coord->J = hthe / Bpxy;
+
+    coord->g_11 = 1.0 / coord->g11 + SQ(I * Rxy);
+    coord->g_22 = SQ(coord->Bxy * hthe / Bpxy);
+    coord->g_33 = Rxy * Rxy;
+    coord->g_12 = Btxy * hthe * I * Rxy / Bpxy;
+    coord->g_13 = I * Rxy * Rxy;
+    coord->g_23 = Btxy * hthe * Rxy / Bpxy;
+
+    coord->geometry();
+
+    /**************** SET EVOLVING VARIABLES *************/
+
+    // Tell BOUT++ which variables to evolve
+    // add evolving variables to the communication object
+    if (evolve_rho) {
+      solver->add(rho, "rho");
+      comms.add(rho);
+      output.write("rho\n");
+    } else
```

Comment:
warning: statement should be inside braces [readability-braces-around-statements]

```cpp
} else
      ^
       {
```